src/posts/posts.service.ts

```diff
@@ -62,10 +65,13 @@ export class PostsService {
         }
         if(post.archived === true){
             if(roleId === 2 ||userId === post.authorId ){
-                return post
+                const action = await this.actionsService.addAction('Viewed',userId,'Post',post.id,post)
```

Comment:
here you have hadcoded strings. Store them in variables and reuse? insted of hardcode them